cloud-yarn/src/main/java/com/sequenceiq/cloudbreak/cloud/yarn/loadbalancer/service/launch/LoadBalancerLaunchService.java

```diff
@@ -0,0 +1,259 @@
+            createApplication(yarnClient, createApplicationRequest);
+        }
+
+        // Create an object for the new loadbalancer application and persist it in the resources table.
+        CloudResource loadBalancerApplication = new Builder().type(YARN_LOAD_BALANCER).name(applicationName).build();
+        persistenceNotifier.notifyAllocation(loadBalancerApplication, authenticatedContext.getCloudContext());
+        return loadBalancerApplication;
+    }
+
+    /**
+     * Creates a load balancer application request, specifically by:
+     *      - Getting the IP addresses of the gateway nodes that are already running
+     *      - Creating YarnComponent objects for each (if more than one) loadbalancer and pointing them to the gateway IPs.
+     */
+    private CreateApplicationRequest createLoadBalancerRequest(CloudStack cloudStack, String applicationName, AuthenticatedContext authenticatedContext,
+            YarnClient yarnClient) {
+        CreateApplicationRequest createApplicationRequest = initializeRequest(cloudStack, applicationName, defaultQueue, defaultLifeTime);
+        List<String> gatewayIPs = getGatewayIPs(authenticatedContext, yarnClient, cloudStack);
+        List<YarnComponent> loadBalancerComponents = createLoadBalancerComponents(cloudStack, gatewayIPs, applicationName);
+        createApplicationRequest.setComponents(loadBalancerComponents);
+        return createApplicationRequest;
+    }
+
+    /**
+     * Uses the provided YarnClient to obtain information about the existing Yarn application. Takes advantage of the fact
+     * that the applicationNameUtil has the exact same logic for creating the name of the original application.
+     *
+     * Returns the extracted IPs of only the gateway nodes in the Yarn application.
+     */
+    private List<String> getGatewayIPs(AuthenticatedContext authenticatedContext, YarnClient yarnClient, CloudStack cloudStack) {
+        LOGGER.debug("Getting the IPs of the existing gatway Yarn containers for the loadbalancer.");
+        String applicationName = applicationNameUtil.createApplicationName(authenticatedContext);
+        Iterable<Container> foundContainers = getContainers(applicationName, yarnClient);
+        Set<String> gatewayGroupNames = getGatewayGroupNames(cloudStack);
+
+        List<String> gatewayIPs = Lists.newArrayList();
+        foundContainers.forEach(container -> {
+            if (gatewayGroupNames.contains(container.getComponentName())) {
+                gatewayIPs.add(container.getIp());
+            }
+        });
+        return gatewayIPs;
+    }
+
+    /**
+     * Gets all of the containers currently running in the YCloud ecosystem within the application
+     * given by the application name provided.
+     */
+    public static Iterable<Container> getContainers(String applicationName, YarnClient yarnClient) {
```

Comment:
Since `LoadBalancerLaunchService` is an injectable object, don't make this static. Instead just inject `LoadBalancerLaunchService` where this needs to be called, and call it through that object.

Reply:
Ahh that makes sense, sorry still getting used to Spring stuff.